core/shared/src/main/scala/monocle/Lens.scala

```diff
@@ -308,4 +308,8 @@ final case class LensSyntax[S, A](private val self: Lens[S, A]) extends AnyVal {
 
   def at[I, A1](i: I)(implicit evAt: At[A, I, A1]): Lens[S, A1] =
     self.andThen(evAt.at(i))
+
+  @deprecated("Preserved for bincompat", "3.1.0")
+  def index[I, A1](i: I, evIndex: Index[A, I, A1]): Optional[S, A1] =
+    self.index(i)(evIndex, implicitly, implicitly)
```

Comment:
This and similar changes fixes bincompat broken in:
- https://github.com/optics-dev/Monocle/pull/1192